src/Adapter/BaseAdapter.php

```diff
@@ -96,4 +103,25 @@ protected function getTimezoneOffset()
 
         return sprintf('%s%d:%02d', $prefix, $hours, $minutes);
     }
+
+    /**
+     * Validate the given select's order by rules
+     *
+     * @param Select $select
+     *
+     * @return void
+     */
+    protected function validateOrderBy(Select $select): void
```

Comment:
I would have the method return bool so that there is one place that throws an exception and not multiple, since we have adapters that need to validate differently. You can also use the `OrderByInterface` as type here. If you don't want to return `bool` please add `@throws` and remove `@return`.

Reply:
Returning a bool doesn't allow for a specific exception message.

Reply:
Yes, I thought so too, but I didn't find it too useful. But as I said, either is fine with me.

Reply:
Well, either that or a specific exception type.

Reply:
The specific exception type is fine. Don't forget to use the `OrderByInterface` 😉.